### PR TITLE
Use shared tsconfig

### PR DESCRIPTION
**Pull request by aaomidi**

### CODE REVIEW

**Comment by aaomidi** [comment]
I'll make PRs after this slowly enabling these rules on these extensions that are not using the rules given by the shared tsconfig.

File: extensions/schema-compare/tsconfig.json [hunk 1]
```diff
@@ -1,8 +1,7 @@
 {
+	"extends": "../shared.tsconfig.json",
```
Comment (udeeshagautam) [inline extensions/schema-compare/tsconfig.json]: extensions/dacpac too please...

Comment (aaomidi) [reply extensions/schema-compare/tsconfig.json]: That's already using the shared config file

**Comment by Charles-Gagnon** [comment]
Glad to see that the node download issues are still happening....

```
Downloading: https://nodejs.org/dist/v10.15.1/node-v10.15.1-darwin-x64.tar.gz
##[error]Unexpected HTTP response: 522
##[section]Finishing: NodeTool
```

File: extensions/shared.tsconfig.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
 		"noImplicitReturns": true,
 		"noUnusedLocals": true,
 		"noUnusedParameters": true,
-		"forceConsistentCasingInFileNames": true
+		"forceConsistentCasingInFileNames": true,
+		"sourceMap": true
```
Comment (anthonydresser) [inline extensions/shared.tsconfig.json]: don't need this, this is ignored during compilation anyways

Comment (aaomidi) [reply extensions/shared.tsconfig.json]: Which option? sourceMap or forceConsistentCasing?

Comment (anthonydresser) [reply extensions/shared.tsconfig.json]: sourcemaps, we use gulp-sourcemaps to generate them and it just ignores this setting. extensions will get sourcemaps without this setting. so there's no need to introduce the merge conflict.

**Pull request merged**